GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift

```diff
@@ -0,0 +1,140 @@
+        
+        XCTAssertTrue(viewController.presentedViewController is CallVisualizer.VisitorCodeViewController)
+    }
+
+    func test_handleAcceptedUpgrade() {
+        var calledEvents: [CallVisualizer.Coordinator.DelegateEvent] = []
+        coordinator.environment.eventHandler = { calledEvents.append($0) }
+        coordinator.handleAcceptedUpgrade()
+
+        XCTAssertTrue(calledEvents.contains(.maximized))
+    }
+
+    func test_handleEngagementRequestAccepted() throws {
+        let site = CoreSdkClient.Site(
+            id: .mock, defaultOperatorPicture: nil,
+            alwaysUseDefaultOperatorPicture: false,
+            allowedFileSenders: try .mock(),
+            maskingRegularExpressions: [],
+            visitorAppDefaultLocale: "",
+            mobileConfirmDialogEnabled: false,
+            mobileObservationIndicationEnabled: true,
+            mobileObservationVideoFps: try videoFps(),
+            mobileObservationEnabled: true
+        )
+
+        coordinator.environment.fetchSiteConfigurations = { callback in
+            callback(.success(site))
+        }
+
+        var answers: [Bool] = []
+        let answer = Command<Bool> { boolean in
+            answers.append(boolean)
+        }
+
+        coordinator.handleEngagementRequestAccepted(answer)
+        XCTAssertEqual(answers, [true])
+    }
+
+    func test_handleEngagementRequestAcceptedMobileConfirmDialogEnabled() throws {
+        let scene = try XCTUnwrap(UIApplication.shared.connectedScenes.first as? UIWindowScene)
+        let window = scene.windows.first
+        let oldRootViewController = window?.rootViewController
+        window?.rootViewController = viewController
+        defer { window?.rootViewController = oldRootViewController }
+
+        let presenter = CallVisualizer.Presenter(presenter: { self.viewController })
+        coordinator.environment.presenter = presenter
+
+        let site = CoreSdkClient.Site(
+            id: .mock, defaultOperatorPicture: nil,
+            alwaysUseDefaultOperatorPicture: false,
+            allowedFileSenders: try .mock(),
+            maskingRegularExpressions: [],
+            visitorAppDefaultLocale: "",
+            mobileConfirmDialogEnabled: true,
+            mobileObservationIndicationEnabled: true,
+            mobileObservationVideoFps: try videoFps(),
+            mobileObservationEnabled: true
+        )
+
+        coordinator.environment.fetchSiteConfigurations = { callback in
+            callback(.success(site))
+        }
+
+        let answer = Command<Bool> { _ in }
+        coordinator.handleEngagementRequestAccepted(answer)
+
+        XCTAssertTrue(coordinator.environment.presenter.getInstance()?.presentedViewController is AlertViewController)
+    }
+
+    func test_end() {
+        var isCalled = false
+        
+        coordinator.environment.screenShareHandler.stop = { _ in
+            isCalled = true
+        }
```

Comment:
would it be beneficial to have a counter here to know about multiple calls if this happens?
```swift
        var stopCallCounter = 0
        
        coordinator.environment.screenShareHandler.stop = { _ in
            stopCallCounter += 1
        }
        // ...
        XCAssertTrue(stopCallCounter)
```

Reply:
Fixed 52f4f8d4ae0dc6303b92c6934efda5240eeaa6e2